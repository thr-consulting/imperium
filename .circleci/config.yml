version: 2

defaults: &defaults
  working_directory: ~/repo
  docker:
    - image: circleci/node:12.7

jobs:
  test:
    <<: *defaults
    steps:
      - checkout

      - restore_cache:
          keys:
            - v1-dependencies-{{ checksum "yarn.lock" }}
            - v1-dependencies-

      - run: yarn global add lerna
      - run: yarn lerna bootstrap

      - save_cache:
          paths:
            - node_modules
          key: v1-dependencies-{{ checksum "yarn.lock" }}

      - run: yarn build
      - run: yarn ts
      - run: yarn lint
      - run: yarn test

      - persist_to_workspace:
          root: ~/repo
          paths: .

  publish:
    <<: *defaults
    steps:
      #      - checkout
      - attach_workspace:
          at: ~/repo

      - run: yarn global add lerna
      - run: echo "//registry.npmjs.org/:_authToken=$npm_TOKEN" > ~/repo/.npmrc
      - run: yarn lerna publish patch -c --preid alpha --yes
      - run: DISCORD_WEBHOOK=$DISCORD_WEBHOOK bash ./tools/discordhook.sh

workflows:
  version: 2
  test-and-publish:
    jobs:
      - test
      - publish:
          requires:
            - test
          filters:
            branches:
              only: master
