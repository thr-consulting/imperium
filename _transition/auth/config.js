/* eslint-disable @typescript-eslint/no-var-requires */
const {name} = require('./package.json');

module.exports = function() {
	return {
		name,
		initialConfig: {
			authAccessTokenLsName: process.env.AUTH_ACCESS_TOKEN_LS_NAME || 'IMP.access',
			authRefreshTokenLsName: process.env.AUTH_REFRESH_TOKEN_LS_NAME || 'IMP.refresh',
		},
	};
};
