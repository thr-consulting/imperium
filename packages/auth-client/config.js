/* eslint-disable @typescript-eslint/no-var-requires */
const {name} = require('./package.json');

module.exports = function() {
	const proto = process.env.SERVER_PROTOCOL || 'http';
	const host = process.env.SERVER_HOST || 'localhost';
	const port = parseInt(process.env.SERVER_PORT || '4001', 10);
	const loginUrl = process.env.AUTH_LOGIN_URL || '/api/login';
	const refreshUrl = process.env.AUTH_REFRESH_URL || '/api/refresh';
	const forgotPasswordUrl = process.env.AUTH_FORGOTPASSWORD_URL || '/api/forgot-password';

	const authLoginUrl = `${proto}://${host}${port === 80 ? '' : `:${port}`}${loginUrl}`;
	const authRefreshUrl = `${proto}://${host}${port === 80 ? '' : `:${port}`}${refreshUrl}`;
	const authForgotPasswordUrl = `${proto}://${host}${port === 80 ? '' : `:${port}`}${forgotPasswordUrl}`;

	return {
		name,
		initialConfig: {
			authLoginUrl,
			authRefreshUrl,
			authForgotPasswordUrl,
			authLSAccessTokenKey: process.env.AUTH_ACCESS_TOKEN_KEY || 'access',
			authLSIdKey: process.env.AUTH_ID_KEY || 'id',
		},
	};
};
