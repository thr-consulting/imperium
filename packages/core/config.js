const path = require('path');

/**
 * This is where the configuration resides in the host project.
 * It is a relative folder from the host project root.
 * @type {string}
 */
const projectData = 'src/imperium';

/**
 * These configuration values tells the Imperium framework how the host project is configured.
 */
module.exports = {
	production: {
		// Where the production host project is built, relative to host project root.
		buildDir: 'build',
		// Should Imperium minimize the host project when built
		minimize: true,
		// Where the webpack report is generated.
		reportFilename: 'report.html',
	},
	project: {
		// The path of the file where the data connectors are defined
		Connectors: path.join(projectData, 'Connectors.js'),
		// The path of the file where the server modules are defined. See (docs/ServerModule.md)
		serverModules: path.join(projectData, 'serverModules.js'),
		// The path of the file where the client modules are defined. See (docs/ClientModule.md)
		clientModules: path.join(projectData, 'clientModules.js'),
		// The path of the file where the default route options are configured.
		routeDefaults: path.join(projectData, 'routeDefaults.js'),
		// The path of the file where the root component is rendered from
		rootRender: path.join(projectData, 'rootRender.js'),
		// The path of the file where options are found for HTML index generation (optional)
		htmlOptions: path.join(projectData, 'htmlOptions.js'),
	},
};
