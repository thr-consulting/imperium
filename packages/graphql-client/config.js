/* eslint-disable @typescript-eslint/no-var-requires */
const {inspectLoader} = require('@imperium/util');
const {name} = require('./package.json');

module.exports = function() {
	const proto = process.env.SERVER_PROTOCOL || 'http';
	const host = process.env.SERVER_HOST || 'localhost';
	const port = parseInt(process.env.SERVER_PORT || '4001', 10);
	const url = process.env.GRAPHQL_URL || '/api/graphql';
	const enableGraphqlWs = process.env.GRAPHQL_ENABLE_SUBSCRIPTIONS === 'true';

	return {
		name,
		initialConfig: {
			graphql: `${proto}://${host}:${port}${url}`,
			graphqlws: enableGraphqlWs ? `ws://${host}:${port}${url}` : false,
		},
		webpack: {
			client: {
				rules: [
					{
						test: /\.graphql$/,
						exclude: /node_modules/,
						use: [
							inspectLoader('GRAPHQL'),
							{
								loader: 'graphql-tag/loader',
							},
						],
					},
				],
			},
		},
	};
};
