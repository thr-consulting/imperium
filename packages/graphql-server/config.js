/* eslint-disable @typescript-eslint/no-var-requires */
const {inspectLoader} = require('@thx/common-webpack');
const {name} = require('./package.json');

module.exports = function() {
	return {
		name,
		webpack: {
			server: {
				rules: [
					{
						test: /\.graphqls$/,
						exclude: /node_modules/,
						use: [
							inspectLoader('GRAPHQLS'),
							{
								loader: 'graphql-tag/loader',
							},
						],
					},
				],
			},
		},
	};
};
