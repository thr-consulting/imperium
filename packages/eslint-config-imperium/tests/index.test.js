import config from '../index';

describe('ESLint Config', () => {
	it('should match the config', () => {
		expect(config).toMatchSnapshot();
	});
});
