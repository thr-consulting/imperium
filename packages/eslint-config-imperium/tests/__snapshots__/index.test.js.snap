// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ESLint Config should match the config 1`] = `
Object {
  "env": Object {
    "commonjs": true,
    "es6": true,
    "jest/globals": true,
    "node": true,
  },
  "extends": Array [
    "airbnb-typescript",
    "plugin:react/recommended",
    "plugin:@typescript-eslint/recommended",
    "plugin:prettier/recommended",
  ],
  "globals": Object {
    "document": true,
    "process": false,
    "window": true,
  },
  "parser": "@typescript-eslint/parser",
  "parserOptions": Object {
    "ecmaFeatures": Object {
      "jsx": true,
    },
    "ecmaVersion": 2020,
    "sourceType": "module",
  },
  "plugins": Array [
    "jest",
    "react-hooks",
  ],
  "rules": Object {
    "@typescript-eslint/ban-ts-ignore": Array [
      "off",
    ],
    "@typescript-eslint/explicit-function-return-type": Array [
      "off",
      Object {
        "allowExpressions": true,
        "allowTypedFunctionExpressions": true,
      },
    ],
    "@typescript-eslint/explicit-member-accessibility": Array [
      "off",
    ],
    "@typescript-eslint/indent": Array [
      "off",
    ],
    "@typescript-eslint/interface-name-prefix": Array [
      "off",
    ],
    "@typescript-eslint/no-before-define": Array [
      "off",
    ],
    "@typescript-eslint/no-explicit-any": "warn",
    "@typescript-eslint/no-unused-expressions": Array [
      "off",
    ],
    "@typescript-eslint/no-unused-vars": Array [
      "error",
      Object {
        "argsIgnorePattern": "server|context|ctx|type",
        "varsIgnorePattern": "d",
      },
    ],
    "class-methods-use-this": "off",
    "global-require": "error",
    "import/extensions": "off",
    "import/no-extraneous-dependencies": "off",
    "import/no-named-default": "off",
    "import/no-unresolved": "off",
    "import/prefer-default-export": "off",
    "jsx-a11y/anchor-is-valid": "off",
    "jsx-a11y/label-has-associated-control": "off",
    "jsx-a11y/label-has-for": "off",
    "lines-between-class-members": Array [
      "error",
      "always",
      Object {
        "exceptAfterSingleLine": true,
      },
    ],
    "no-console": Array [
      "error",
    ],
    "no-plusplus": "off",
    "no-return-assign": Array [
      "error",
      "except-parens",
    ],
    "no-underscore-dangle": Array [
      "warn",
      Object {
        "allow": Array [
          "_id",
        ],
        "allowAfterThis": true,
      },
    ],
    "no-unused-vars": Array [
      "error",
      Object {
        "argsIgnorePattern": "server|context|ctx|type",
        "varsIgnorePattern": "d",
      },
    ],
    "no-use-before-define": Array [
      "off",
    ],
    "prefer-arrow-callback": Array [
      "error",
      Object {
        "allowNamedFunctions": true,
      },
    ],
    "react-hooks/exhaustive-deps": "warn",
    "react-hooks/rules-of-hooks": "error",
    "react/destructuring-assignment": Array [
      "off",
      "always",
    ],
    "react/display-name": Array [
      "off",
    ],
    "react/forbid-prop-types": "error",
    "react/jsx-curly-newline": Array [
      "off",
    ],
    "react/jsx-filename-extension": Array [
      "error",
      Object {
        "extensions": Array [
          ".tsx",
        ],
      },
    ],
    "react/jsx-indent": Array [
      "off",
    ],
    "react/jsx-indent-props": Array [
      "off",
    ],
    "react/jsx-one-expression-per-line": "off",
    "react/jsx-props-no-spreading": Array [
      "off",
    ],
    "react/jsx-wrap-multilines": Array [
      "off",
    ],
    "react/no-children-prop": "off",
    "react/no-typos": "off",
    "react/no-unknown-property": Array [
      "error",
      Object {
        "ignore": Array [
          "for",
        ],
      },
    ],
    "react/no-unused-prop-types": "off",
    "react/prefer-stateless-function": "error",
    "react/require-default-props": "off",
  },
  "settings": Object {
    "react": Object {
      "version": "detect",
    },
  },
}
`;
